src/coreclr/Directory.Build.props

```diff
@@ -1,28 +1,168 @@
 <Project>
 
+  <Import Project="..\..\Directory.Build.props" />
+
+  <!-- Set default Configuration and Platform -->
   <PropertyGroup>
-    <SkipImportArcadeSdkFromRoot>true</SkipImportArcadeSdkFromRoot>
+    <ArchGroup Condition="'$(__BuildArch)' != ''">$(__BuildArch)</ArchGroup>
+    <ArchGroup Condition="'$(ArchGroup)' == ''">x64</ArchGroup>
+    <ArchGroup Condition="'$(ArchGroup)' == 'amd64'">x64</ArchGroup>
+    <ArchGroup Condition="'$(ArchGroup)' == 'armel'">arm</ArchGroup>
+    <TargetArchitecture Condition="'$(TargetArchitecture)' == ''">$(ArchGroup)</TargetArchitecture>
+
+    <OSGroup Condition="'$(__BuildOS)' != ''">$(__BuildOS)</OSGroup>
+    <OSGroup Condition="'$(OSGroup)' == '' and '$([MSBuild]::IsOSPlatform(Windows))' == 'true'">Windows_NT</OSGroup>
+    <OSGroup Condition="'$(OSGroup)' == '' and '$([MSBuild]::IsOSPlatform(Linux))' == 'true'">Linux</OSGroup>
+    <OSGroup Condition="'$(OSGroup)' == '' and '$([MSBuild]::IsOSPlatform(OSX))' == 'true'">OSX</OSGroup>
+    <OSGroup Condition="'$(OSGroup)' == ''">AnyOS</OSGroup>
   </PropertyGroup>
-  <Import Project="..\..\Directory.Build.props" />
 
-  <!-- Ensure our properties are set before Arcade defines defaults -->
-  <Import Project="dir.common.props" />
-  <Import Project="Sdk.props" Sdk="Microsoft.DotNet.Arcade.Sdk" />
+  <!--
+    If Configuration is empty that means we are not being built in VS and so folks need to explicitly pass the different
+    values for $(ConfigurationGroup) or $(OSGroup) or accept the defaults for them.
+  -->
+  <PropertyGroup Condition="'$(Configuration)' == ''">
+    <ConfigurationGroup Condition="'$(__BuildType)' != ''">$(__BuildType)</ConfigurationGroup>
+    <ConfigurationGroup Condition="'$(ConfigurationGroup)' == '' or '$(ConfigurationGroup)' == 'debug'">Debug</ConfigurationGroup>
+    <ConfigurationGroup Condition="'$(ConfigurationGroup)' == 'release'">Release</ConfigurationGroup>
+    <ConfigurationGroup Condition="'$(ConfigurationGroup)' == 'checked'">Checked</ConfigurationGroup>
```

Comment:
Can't we use ToUpper here?

Reply:
We could, but ToUpper converts all letters. We just want to upper-case the first one so I think this is the easiest solution in msbuild. Also this is just a move.

Reply:
I thought there was an API to just upper case the first letter, but that's fine.

Reply:
I couldn't find it, does one exist?

Reply:
Hmm it seems like there's not one for such operation, I thought there was in C#, apparently not.